src/pytorch_lightning/plugins/precision/fsdp_native_native_amp.py

```diff
@@ -25,14 +25,13 @@
     from torch.distributed.fsdp.sharded_grad_scaler import ShardedGradScaler
 else:
     MixedPrecision = None  # type: ignore[misc,assignment]
+    ShardedGradScaler = None  # type: ignore[misc,assignment]
 
 
 class FullyShardedNativeNativeMixedPrecisionPlugin(NativeMixedPrecisionPlugin):
     """Native AMP for Fully Sharded Native Training."""
 
-    def __init__(
-        self, precision: Union[str, int], device: str, scaler: Optional[torch.cuda.amp.GradScaler] = None
-    ) -> None:
+    def __init__(self, precision: Union[str, int], device: str, scaler: Optional[ShardedGradScaler] = None) -> None:
```

Comment:
The `ShardedGradScaler` subclasses the `cuda.amp.GradScaler`. Did you change this because mypy was complaining?

Does the code require this specific subclass?

If yes, then the annotation for `src/pytorch_lightning/plugins/precision/sharded_native_amp.py` should also be updated.

Reply:
no, mypy wasn't failing. I just wanted to be explicit to make sure users passes the `ShardedGradScaler` with shared strategies. If you agree on this, will update `sharded_native_amp` too else revert this.

Reply:
It's a bit restrictive but I'm fine with doing it for clarity. 

> will update sharded_native_amp

:+1: